src/osemgrep/cli_scan/Scan_subcommand.ml

```diff
@@ -580,7 +580,15 @@ let run_scan_files (_caps : < Cap.stdout >) (conf : Scan_CLI.conf)
         || Output_format.keep_ignores output_format
       in
       let filtered_matches =
-        Nosemgrep.filter_ignored ~keep_ignored res.core.results
+        let results =
+          List_.map
```

Comment:
Move this code in a separate helper function,
maybe you can move this helper in src/fixing/Autofix.ml maybe with a nice comment in the .ml or .mli explaining why we need that.

Reply:
done in [33d3ea0](https://github.com/semgrep/semgrep/pull/9645/commits/33d3ea0b95afd1134fc271b62938230d330e7526) -- moved to a helper function and wrote a comment. I'm still not sure why this is needed, and eventually this should be tracked down to where "fix" is created (somewhere in semgrep-core).. -- I had a quick look at it, but wasn't able to spot anything obvious